ui/index.d.ts

```diff
@@ -1251,13 +1251,21 @@ export function createAzExtOutputChannel(name: string, extensionPrefix: string):
  */
 export function openReadOnlyJson(node: { label: string, fullId: string }, data: {}): Promise<void>;
 
+export class ReadOnlyContent {
+    public append(content: string): Promise<void>;
+    public clear(): void;
+    public isVisible(): Promise<boolean>;
+    public show(options?: TextDocumentShowOptions): Promise<void>;
```

Comment:
Not 100% convinced `isVisible` and `show` are needed for your current scenario, but could see them being useful in the future so 🤷‍♂️